### PR TITLE
feat(gomod): Add `gomodSkipVendor` postUpdateOption

### PR DESCRIPTION
**Pull request by r0bobo**
## Changes

This PR adds a postUpdateOption `gomodSkipVendor` which disables automatic vendoring functionality in the `gomod` manager.

## Context

I'm creating this PR because I've had the issue described in:
- #9271

Files that are removed by `go mod vendor` but are then restored in a `postUpgradeTask` are not added to the final commit.
After finding [this comment](https://github.com/renovatebot/renovate/discussions/29107#discussioncomment-9459541) in a discussion about go vendoring trouble I took a stab at adding the suggested option.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

**Review by rarkins — CHANGES_REQUESTED** [review]
Add a test for this skip case

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -2508,6 +2508,7 @@ const options: RenovateOptions[] = [
       'gomodTidy1.17',
       'gomodTidyE',
       'gomodUpdateImportPaths',
+      'gomodVendorSkip',
```
Comment (viceice) [inline lib/config/options/index.ts:2511]:
```suggestion
      'gomodSkipVendor',
```
sounds better to me

File: lib/modules/manager/gomod/artifacts.ts [hunk 2]
```diff
@@ -133,7 +133,9 @@ export async function updateArtifacts({
 
   const vendorDir = upath.join(goModDir, 'vendor/');
   const vendorModulesFileName = upath.join(vendorDir, 'modules.txt');
-  const useVendor = (await readLocalFile(vendorModulesFileName)) !== null;
+  const useVendor =
+    (await readLocalFile(vendorModulesFileName)) !== null &&
+    !config.postUpdateOptions?.includes('gomodVendorSkip');
```
Comment (viceice) [inline lib/modules/manager/gomod/artifacts.ts:138]: invert condition

Comment (r0bobo) [reply lib/modules/manager/gomod/artifacts.ts:138]: Do you mean as in first checking if the option is not set and then checking the file, or as in flipping the meaning of the variable to something like `skipVendor`?

Comment (viceice) [reply lib/modules/manager/gomod/artifacts.ts:138]: yes, first checking the option. that should be faster and we don't need to read the file if we skip anyways

Comment (r0bobo) [reply lib/modules/manager/gomod/artifacts.ts:138]: Right, I'll fix it.

**Comment by r0bobo** [comment]
> Add a test for this skip case

I added a test that copies the `it('supports vendor directory update', ...)` case but passes the `gomodSkipVendor` as well. I started editing the expected output so that the diff would disappear, but I kinda feel like I don't know what I'm doing as I don't understand the testing framework that well. Do you have any pointers on how the test case should look?

**Review by rarkins — DISMISSED** [review]

**Comment by r0bobo** [comment]
I've fixed the test so it is now green.

**Review by rarkins — APPROVED** [review]

**Pull request merged**